src/dotnet/Program.cs

```diff
@@ -253,6 +255,27 @@ private static bool IsDotnetBeingInvokedFromNativeInstaller(TopLevelCommandParse
             }
         }
 
+        internal static void InitializeEnvironment()
+            => InitializeEnvironment(new EnvironmentProvider());
+
+        internal static void InitializeEnvironment(IEnvironmentProvider env)
+        {
+            var dotnetRootEnv = env.Is64BitProcess
+                ? "DOTNET_ROOT"
+                : "DOTNET_ROOT(x86)";
+
+            try
+            {
+                var rootPath = Path.GetDirectoryName(new Muxer().MuxerPath);
+                env.SetEnvironmentVariable(dotnetRootEnv, rootPath, EnvironmentVariableTarget.Process);
```

Comment:
This will fix some cases and break others. 

Consider either of these:
1. DOTNET_ROOT was already set to a location with runtimes needed for my tools,
2. DOTNET_ROOT was not set, but default global location has the runtimes needed for my tools

When the unzipped CLI does not have the appropriate runtimes.

Example (not-so-contrived IMHO) case:

* Tools I use are all stable and using .NET Core 2.x; 2.x runtimes
* I unzip the .NET Core 3.x SDK which only has the 3.x runtime

AFAICT, this change doesn't even give me a workaround if I'm broken by it.

Reply:
Arguments for or against this fix aside, we should only be setting the variable if not already present to allow users to have some control over the value.

Reply:
While I generally agree with the principle, I'm not convinced there are scenarios where we would want to let the user direct DOTNET_ROOT away from the SDK they are invoking. It seems like this will lead to more problems, and it's unclear to me it solves any valuable scenarios.

Reply:
What if the `dotnet` being invoked isn't on `PATH` and also l locally installed?

Reply:
Erm, never mind, that's not a problem since we're getting the path to the command here from the muxer.  I think you're right that it probably isn't a useful scenario, I just don't like tools that use environment variables but don't allow me to control them.

Reply:
Personal feelings about respecting externally set environment variables aside, after more thought (and poking around with how our muxer resolution actually works in the CLI), I can't think of a reason to respect an already set `DOTNET_ROOT` that would point at something other than the currently running `dotnet`'s directory (assuming, of course, that muxer resolution was successful).